src/crypto/eax.js

```diff
@@ -0,0 +1,162 @@
+// OpenPGP.js - An OpenPGP implementation in javascript
+// Copyright (C) 2018 ProtonTech AG
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 3.0 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+/**
+ * @fileoverview This module implements AES-EAX en/decryption on top of
+ * native AES-CTR using either the WebCrypto API or Node.js' crypto API.
+ * @requires asmcrypto.js
+ * @requires crypto/cmac
+ * @requires util
+ * @module crypto/eax
+ */
+
+import { AES_CTR } from 'asmcrypto.js/src/aes/ctr/exports';
+import CMAC from './cmac';
+import util from '../util';
+
+const webCrypto = util.getWebCryptoAll();
+const nodeCrypto = util.getNodeCrypto();
+const Buffer = util.getNodeBuffer();
+
+
+const blockLength = 16;
+const ivLength = blockLength;
+const tagLength = blockLength;
```

Comment:
This makes sense to me intuitively, but do you have any idea why the paper uses a variable tag length? I could not figure that out. But if there is a good reason then maybe we should allow this to be a configuration option.

Reply:
No — I suspect they found this more elegant/general. I don't see a good reason to use a shorter one unless you have very strict bandwidth/storage and loose authenticity requirements. RFC4880bis-04 currently mandates a tag length of 16, but I guess we could make it an option if we ever want to use the eax implementation for something else.